### PR TITLE
fix kubectl printer flag

### LINKED ISSUE #174: deprecate the print flag `--template`
**What keywords did you search in Kubernetes issues before filing this one?** :
'template'  'printer'  'kubectl'
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

FEATURE REQUEST

**What happened**:

Recently I was learning code about printer. Little confused about the flag `--template`.
The flag `--template` have a description: "Template string or path to template file to use when -o=go-template, -o=go-template-file."
But base the code:
```go
	if flag := flags.Lookup("template"); flag != nil {
		if flag.Value.Type() == "string" {
			templateFile = GetFlagString(cmd, "template")
		}
	}
	if len(outputFormat) == 0 && len(templateFile) != 0 {
		outputFormat = "template"
	}
```
This flag be useful only if don't have `-o` flag.

And also, this flag have the same function with `-o go-template=****`. I'm not sure why we have this flag, is there any other reasons I don't know? If so, I think we should at least change the description of this flag.

Besides, since `-o template\-o templatefile` is same with `-o go-template\-o go-template-file`, can we chose one? There don't have any hint in `-o` flag about `-o template\-o templatefile` .

**What you expected to happen**:
1. Maybe deprecate the flag `--template` if we can make sure this flag is useless.
2. Change the description of flag `--template` if it still useful.
3. Maybe chose one in `-o template`\`-o templatefile` and  `-o go-template`\`-o go-template-file`.

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by Lion-Wei**
**What this PR does / why we need it**:

Make the description of kubectl `--template` flag more accurate, and delete `--templatefile` flag, since it is basically useless.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubectl#174

**Special notes for your reviewer**:

Actually I found `--template` and `--templatefile` have the same function with `-o go-template` and `-o go-template-file`, and the latter are more consistent with other print options. I think if don't have other consideration, I think the flag `--templatefile` and `--template` can be deprecated.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by Lion-Wei** [comment]
@apelisse @mengqiy Can you help me take a look?
/assign @smarterclayton

File: pkg/kubectl/cmd/util/printing.go
Comment (MHBauer) [inline pkg/kubectl/cmd/util/printing.go]: `have no` -> `there is no`

**Review by MHBauer — CHANGES_REQUESTED** [review]
suggested fix for the phrasing.
I cannot comment on the correctness of the change. Referenced issue is good resource for why this change is needed.

File: pkg/kubectl/cmd/util/printing.go
Comment (MHBauer) [reply pkg/kubectl/cmd/util/printing.go]: alternatively, `when no -o flag is provided`

Comment (Lion-Wei) [reply pkg/kubectl/cmd/util/printing.go]: Thanks.

File: pkg/printers/printers.go
Comment (smarterclayton) [inline pkg/printers/printers.go]: Why is this being removed?  If it's possible for a user to specify this, this would not be a backwards compatible change on the CLI.

Comment (Lion-Wei) [reply pkg/printers/printers.go]: I remove it because I didn't saw this output option in `kubectl get -h` or anywhere, but if for backwards compatible, that's make sense.
Sorry I'm not familiar with the early version.

Comment (Lion-Wei) [reply pkg/printers/printers.go]: Fixed, @smarterclayton  : )

**Comment by Lion-Wei** [comment]
@mengqiy @shiywang Friendly ping. : )

**Comment by Lion-Wei** [comment]
@mengqiy @shiywang .Friendly ping for review. : )

File: pkg/kubectl/cmd/util/printing.go
Comment (mengqiy) [inline pkg/kubectl/cmd/util/printing.go]: https://github.com/kubernetes/kubernetes/issues/12668 has not been fixed yet.

**Comment by mengqiy** [comment]
/assign @juanvallejo 
PTAL to make sure it make sense to you.

**Comment by juanvallejo** [comment]
lgtm pending @mengqiy 's review comment

File: pkg/kubectl/cmd/util/printing.go
Comment (soltysh) [reply pkg/kubectl/cmd/util/printing.go]: I find the previous text a better fit, the proposed change is confusing.

**Review by soltysh — CHANGES_REQUESTED** [review]
/hold

**Comment by Lion-Wei** [comment]
@soltysh Yeah,,, I found this pr is not appropriate too...
I think it's better be closed, sorry guys. : (
 /close

**Pull request closed without merging**
